ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/deneb/util/AttestationUtilDeneb.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright ConsenSys Software Inc., 2023
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.logic.versions.deneb.util;
+
+import static tech.pegasys.teku.infrastructure.time.TimeUtilities.millisToSeconds;
+import static tech.pegasys.teku.infrastructure.time.TimeUtilities.secondsToMillis;
+
+import java.util.Optional;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.config.SpecConfig;
+import tech.pegasys.teku.spec.datastructures.operations.Attestation;
+import tech.pegasys.teku.spec.logic.common.helpers.BeaconStateAccessors;
+import tech.pegasys.teku.spec.logic.common.helpers.MiscHelpers;
+import tech.pegasys.teku.spec.logic.versions.altair.util.AttestationUtilAltair;
+import tech.pegasys.teku.spec.schemas.SchemaDefinitions;
+
+public class AttestationUtilDeneb extends AttestationUtilAltair {
+
+  public AttestationUtilDeneb(
+      final SpecConfig specConfig,
+      final SchemaDefinitions schemaDefinitions,
+      final BeaconStateAccessors beaconStateAccessors,
+      final MiscHelpers miscHelpers) {
+    super(specConfig, schemaDefinitions, beaconStateAccessors, miscHelpers);
+  }
+
+  /**
+   * [IGNORE] attestation.data.slot is equal to or earlier than the current_slot (with a
+   * MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance) -- i.e. attestation.data.slot <= current_slot (a
+   * client MAY queue future attestation for processing at the appropriate slot).
+   *
+   * <p>[IGNORE] the epoch of attestation.data.slot is either the current or previous epoch (with a
+   * MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance) -- i.e. compute_epoch_at_slot(attestation.data.slot)
+   * in (get_previous_epoch(state), get_current_epoch(state))
+   */
+  @Override
+  public Optional<SlotInclusionGossipValidationResult> performSlotInclusionGossipValidation(
+      final Attestation attestation, final UInt64 genesisTime, final UInt64 currentTimeMillis) {
```

Comment:
Do we guard against pre-genesis somewhere ? we'll be pre-genesis a lot in test nets is my main concern...

Reply:
Yeah, when we initialise the store, we set the time to at least genesis and don't update it until we hit it in `StoreBuilder` class:

![image](https://github.com/Consensys/teku/assets/14827647/7fbce9d3-2ec2-4749-b709-4bc16afdaf89)